src/pydvl/valuation/samplers/base.py

```diff
@@ -113,14 +112,33 @@ def from_data(self, data: Dataset) -> BatchGenerator:
 
     def from_indices(self, indices: IndexSetT) -> BatchGenerator:
         """Batches the samples and yields them."""
+
+        # early return for empty indices. Necessary because some samplers use a
+        # while True loop to generate infinite samples.
+        if len(indices) == 0:
+            return
```

Comment:
I added a comment to explain why that is not the case. The reason is that `generate_batches` is a generator function (because it has a yield statement later in its body) and inside generator functions `return` acts more like a `break` statement in a loop. It's described in [this post](https://stackoverflow.com/a/13243870) on stackoverflow.

The simplest example for this behavior is:

```python
>>> def f():
...     return
...     yield 

# f is a generator
>>> type(f())
generator

# the generator is empty
>>> list(f())
[]

The return type is not None
>>> f() is None:
False
```